### PR TITLE
feature: support a proper paginator from Elasticsearch

### PR DESCRIPTION
**Pull request by dkarlovi**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

This PR allows our Elasticsearch extension to return a proper pagination object which will correctly say the total number of items found within Elasticsearch.

Currently, the paginator will always claim there's exactly the number of items which are on the current page (which is a transitive bug caused by the lack of this feature).

### CODE REVIEW

File: lib/ApiPlatform/Elasticsearch/Extension/SearchExtension.php [hunk 1]
```diff
@@ -46,8 +50,26 @@ public function __construct(RequestStack $requestStack, ManagerRegistry $registr
...
+     * @inheritDoc
+     */
+    public function getResult(QueryBuilder $queryBuilder, string $resourceClass = null, string $operationName = null, array $context = [])
+    {
+        return new PrecountedPaginator(new Paginator($queryBuilder), $this->totalItems);
+    }
+
     public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null)
     {
+        $this->totalItems = null;
```
Comment (dkarlovi) [inline lib/ApiPlatform/Elasticsearch/Extension/SearchExtension.php]: Not a fan of this, but currently don't see a better solution.

**Comment by dkarlovi** [comment]
This is done, failures unrelated.

**Comment by sstok** [comment]
Can you add some tests? 

And this https://travis-ci.org/rollerworks/search/jobs/533315055#L783 can be fixed by bumping the API-Platform minimum version.

**Comment by dkarlovi** [comment]
@sstok will work on this Thursday. Other than that, you have no other objections?

Thanks for looking at this BTW. :+1:

**Comment by dkarlovi** [comment]
I forgot about this one :see_no_evil: 

Will need to finish up here soonish.

**Pull request closed without merging**